asynction/security/types.py

```diff
@@ -0,0 +1,190 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Mapping, Optional, Sequence, Type
+
+from svarog import register_forge
+from svarog.types import Forge
+
+from asynction.common_types import JSONMapping
+from .exceptions import UnsupportedSecurityScheme
+
+
+class HTTPSecuritySchemeType(Enum):
+    BASIC = "basic"
+    DIGEST = "digest"
+    BEARER = "bearer"
+
+
+class OAuth2FlowType(Enum):
+    """
+    https://www.asyncapi.com/docs/specifications/v2.2.0#oauthFlowsObject
+    """
+    IMPLICIT = "implicit"
+    PASSWORD = "password"
+    CLIENT_CREDENTIALS = "clientCredentials	"
+    AUTHORIZATION_CODE = "authorizationCode"
+
+
+@dataclass
+class OAuth2Flow:
+    """
+    https://www.asyncapi.com/docs/specifications/v2.2.0#oauthFlowObject
+    """
+    scopes: Sequence[str]
+    authorization_url: Optional[str] = None
+    token_url: Optional[str] = None
+    refresh_url: Optional[str] = None
+
+    @staticmethod
+    def forge(
+        type_: Type["OAuth2Flow"],
+        data: JSONMapping,
+        forge: Forge
+    ) -> "OAuth2Flow":
+        return type_(
+            scopes=forge(
+                type_.__annotations__["scopes"],
+                data.get("scopes")
+            ),
+            authorization_url=forge(
+                type_.__annotations__["authorization_url"],
+                data.get("authorizationUrl")
+            ),
+            token_url=forge(
+                type_.__annotations__["token_url"],
+                data.get("tokenUrl")
+            ),
+            refresh_url=forge(
+                type_.__annotations__["refresh_url"],
+                data.get("refreshUrl")
+            )
+        )
+
+
+register_forge(OAuth2Flow, OAuth2Flow.forge)
+
+
+class SecuritySchemesType(Enum):
+    """
+    https://www.asyncapi.com/docs/specifications/v2.2.0#securitySchemeObject
+    """
+    USER_PASSWORD = "userPassword"
+    API_KEY = "apiKey"
+    X509 = "X509"
+    SYMMETRIC_ENCRYPTION = "symmetricEncryption"
+    ASYMMETRIC_ENCRYPTION = "asymmetricEncryption"
+    HTTP_API_KEY = "httpApiKey"
+    HTTP = "http"
+    OAUTH2 = "oauth2"
+    OPENID_CONNECT = "openIdConnect"
+    PLAIN = "plain"
+    SCRAM_SHA256 = "scramSha256"
+    SCRAM_SHA512 = "scramSha512"
+    GSSAPI = "gssapi"
+
+
+@dataclass
+class SecurityScheme:
+    """
+    https://www.asyncapi.com/docs/specifications/v2.2.0#securitySchemeObject
+    """
+    type: SecuritySchemesType
+    description: Optional[str] = None
+    name: Optional[str] = None  # Required for httpApiKey
+    in_: Optional[str] = None  # Required for httpApiKey | apiKey
+    scheme: Optional[HTTPSecuritySchemeType] = None  # Required for http
+    bearer_format: Optional[str] = None  # Optional for http ("bearer")
+    flows: Optional[Mapping[OAuth2FlowType, OAuth2Flow]] = None  # Required for oauth2
```

Comment:
I do see the reason behind choosing a `Mapping` type of enum keys for the [`OAuth2Flows` object](https://www.asyncapi.com/docs/specifications/v2.2.0#oauthFlowsObject), however I feel that a dataclass would fit better as it allows the use of [`__post_init__` validation](https://docs.python.org/3/library/dataclasses.html#post-init-processing).

For example:

```python
@dataclass
class OAuthFlows:
    """https://www.asyncapi.com/docs/specifications/v2.2.0#oauthFlowsObject"""

    internal: Optional[OAuthFlow] = None
    password: Optional[OAuthFlow] = None
    client_credentials: Optional[OAuthFlow] = None
    authorization_code: Optional[OAuthFlow] = None

    def __post_init__(self):
        if self.internal is not None:
            if self.internal.authorization_url is None:
                raise ValueError("Internal OAuth flow is missing Authorization URL")
        # and similar rules for the rest of the flow kinds, as described in the AsyncAPI docs
```